homework/shared_ptr/test.cpp

```diff
@@ -97,7 +97,56 @@ TEST_F(weakPtrTest, shouldCopyAssign) {
 }
 
 TEST_F(weakPtrTest, testLock) {
-    cs::weak_ptr weak(sPtr);
+    cs::weak_ptr<int> wPtr(sPtr);
+    ASSERT_EQ(*(wPtr.lock()), testValueOne);
+}
+
+TEST_F(weakPtrTest, testCopyConstructorFromShared) {
+    cs::weak_ptr<int> wPtr(sPtr);
+    ASSERT_EQ(*(wPtr.lock()), testValueOne);
+    ASSERT_EQ(wPtr.use_count(), 1);
+}
+
+TEST_F(weakPtrTest, testCopyConstructorFromWeak) {
+    cs::weak_ptr<int> wPtr(sPtr);
+    auto wPtr2(wPtr);
+    ASSERT_FALSE(wPtr.expired());
+    ASSERT_EQ(*(wPtr.lock()), testValueOne);
+    ASSERT_EQ(wPtr.use_count(), 2);
+}
+
+TEST_F(weakPtrTest, testCopyAssignment) {
+    cs::weak_ptr<int> wPtr = sPtr;
+    ASSERT_EQ(*(wPtr.lock()), testValueOne);
+    ASSERT_EQ(wPtr.use_count(), 1);
+}
+
+TEST_F(weakPtrTest, testMoveConstructorFromShared) {
+    cs::weak_ptr<int> wPtr{sPtr};
+    ASSERT_EQ(*(wPtr.lock()), testValueOne);
+    ASSERT_EQ(wPtr.use_count(), 1);
+}
+
+TEST_F(weakPtrTest, testMoveConstructorFromWeak) {
+    cs::weak_ptr<int> wPtr{sPtr};
+    auto wPtr2(std::move(wPtr));
+    ASSERT_EQ(*(wPtr2.lock()), testValueOne);
+    ASSERT_EQ(wPtr2.use_count(), 1);
+    ASSERT_TRUE(wPtr.expired());
+}
+
+TEST_F(weakPtrTest, testMoveAssignment) {
```

Comment:
you don't move shared to weak

Reply:
test corrected
weak constructed from shared
second weak created by move assignment from first weak